packages/terra-button/src/Button.jsx

```diff
@@ -303,11 +303,11 @@ class Button extends React.Component {
 
     return (
       <ComponentType
+        tabIndex={isDisabled ? '-1' : undefined}
```

Comment:
I think the change makes sense. The pattern that we typically follow is to add `tabIndex` as a first class prop and then set the tabIndex if one is provided.

`tabIndex={isDisabled ? '-1' : tabIndex}`

Reply:
Since tabIndex is a normal DOM attr, why would we expose it as a 1st class prop in our component? As a pattern - there are so many DOM attrs we implicitly support, how do we decide which make it into 1st class prop?

Reply:
I don't think we have any hard rules to determine whether or not a prop must be first-class or can just be a passthrough. Generally speaking, we'd want to expose a prop as first-class to consumers when the component is more opinionated about this prop or want to have more control on how this prop is applied. 

One scenario this change may not work as intended is when the consumer passed in a tabIndex of 1 with `isDisabled` set to true. Previously it would set tabIndex to -1 but now it would honor the tabIndex to 1 that is passed in when it is disabled. We also usually spread the customProps (`{...customProps}`) before setting all the first-class props so that these first-class props have higher precedence over all the passthrough props.

Reply:
Generally, Terra recommends that consumers and projects do not set custom tabindex for interactive elements (even though they may be tempted to), as it destroys proper accessibility.

- `-1` on non-disabled + interactive elements is bad, unless for very rare and specific places that only receive programmatic focus, which usually are higher-order components and usages that already have a very keen understanding of proper accessibility details needing to be included (i.e. not general `terra-button` use).
- `0` on <button> and <a> elements is superfluous and redundant, as such, is bad.
- `1` (or any number greater than 1) is bad, as it supersedes the natural flow for DOM order keyboard navigation, and should not be used ever by anyone https://webaim.org/techniques/keyboard/tabindex and violates [jsx-eslint/.../ tabindex-no-positive.md](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/tabindex-no-positive.md) and would need overriding.
  
 \
 \
Which is why this prop was never exposed before as first-class. However, it also was probably why it was never tested via spreading customProps as @benbcai mentioned, to see if we were swallowing it or honoring it if provided.
  
Without very specific reusability reasons, I would prefer rather than supporting this first-class, it would be safer to write separate guide docs for how spreading customProps works, include an example for advanced component use where this specific case is needed for `terra-button`, and explain risks of what can be screwed up if done incorrectly. 

And correct the swallowing of the tabindex value if provided by consumers via customProps and honor it (unless is disabled) by changing to this:

```
        tabIndex={isDisabled ? '-1' : customProps.tabIndex}
```

Reply:
That's a good point @neilpfeiffer. If we don't want consumers to set tabIndex, then we shouldn't make it a first-class prop.